core/tests/protractor_utils/AdminPage.js

```diff
@@ -142,6 +152,61 @@ var AdminPage = function() {
     });
   };
 
+  this.startOneOffJob = function(jobName) {
+    this._startOneOffJob(jobName, 0);
+  };
+
+  this._startOneOffJob = function(jobName, i) {
+    waitFor.visibilityOf(oneOffJobRows.first(),
+      'Starting one off jobs taking too long to appear.');
+    oneOffJobRows.get(i).getText().then((text) => {
+      if (text.toLowerCase().startsWith(jobName.toLowerCase())) {
+        oneOffJobRows.get(i).element(
+          by.css('.protractor-test-one-off-jobs-start-btn')).click();
+      } else {
+        this._startOneOffJob(jobName, ++i);
+      }
+    });
+  };
+
+  this.stopOneOffJob = function(jobName) {
+    this._stopOneOffJob(jobName, 0);
+  };
+
+  this._stopOneOffJob = function(jobName, i) {
+    unfinishedOneOffJobRows.get(i).getText().then((text) => {
+      if (text.toLowerCase().startsWith(jobName.toLowerCase())) {
+        unfinishedOneOffJobRows.get(i).element(
+          by.css('.protractor-test-one-off-jobs-stop-btn')).click();
+      } else {
+        this._stopOneOffJob(jobName, ++i);
+      }
+    });
+  };
+
+  this.expectNumberOfRunningOneOffJobs = function(count) {
+    element.all(by.css(
+      '.protractor-test-unfinished-one-off-jobs-id')).count().then((len) =>{
+      expect(len).toEqual(count);
+    });
+  };
+
+  this.expectJobTobeRunning = function(jobName) {
+    browser.refresh();
+    waitFor.pageToFullyLoad();
+    waitFor.visibilityOf(element(
+      by.css('.protractor-test-unfinished-jobs-card')),
+    'Unfinished Jobs taking too long to appear');
+    let unfinishedJobs = unfinishedOffJobIDs.filter((element) => {
+      return element.getText().then((job) => {
+        return job.toLowerCase().startsWith(jobName.toLowerCase());
+      });
+    });
+    unfinishedJobs.get(0).getText((job) => {
```

Comment:
This looks like it only works if the job to check is at the top of the list. What if it isn't? It seems reasonable for someone to launch two jobs and want to check that they are both running, which I don't think this will work for

Reply:
No that's not the case, observe the code from line 200 to 204, it gives us an array having exactly one element since I am filtering all the unfinished tests to get the one that we want to check.

Reply:
Ahh yes, I see now